### PR TITLE
MPU9250: Calibration Routine Optimization

### PR DESCRIPTION
**Pull request by LETS-BEE**
and fix some register.
and... reduce memory.
mpu9250 main, mpu6500 aux. this can finish the calibration process.

### CODE REVIEW

File: src/sensors/mpu9250sensor.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
     SlimeVR Code is placed under the MIT license
-    Copyright (c) 2021 Eiren Rain, S.J. Remington
+    Copyright (c) 2022 Eiren Rain, S.J. Remington, SlimeVR contributors
```
Comment (TheDevMinerTV) [inline src/sensors/mpu9250sensor.cpp:3]: You can leave the year the same

File: src/sensors/mpu9250sensor.cpp [hunk 2]
```diff
@@ -214,6 +217,44 @@ void MPU9250Sensor::getMPUScaled()
...
+    }
+    m_Logger.debug("Calculating calibration data...");
+
+    float M_BAinv[4][3];
+    CalculateCalibration(calibrationDataMag, calibrationSamples, M_BAinv);
+    free(calibrationDataMag);
+
+    m_Logger.debug("[INFO] Magnetometer calibration matrix:");
+    m_Logger.debug("{");
+    for (int i = 0; i < 3; i++) {
```
Comment (TheDevMinerTV) [inline src/sensors/mpu9250sensor.cpp:248]: Please format, either you put `{` on the next line or on the current line, not both.

File: src/sensors/mpu9250sensor.cpp [hunk 3]
```diff
@@ -168,14 +168,12 @@ void MPU9250Sensor::getMPUScaled()
     float temp[3];
     int i;
     int16_t ax,ay,az,gx,gy,gz,mx,my,mz;
```
Comment (TheDevMinerTV) [inline src/sensors/mpu9250sensor.cpp:170]: You could move these variables (`ax,ay,az,gx,gy,gz,mx,my,mz`) into the `if defined(_MAHONY_H_) || defined(_MADGWICK_H_)` to save even more stack, I believe.

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

File: src/sensors/mpu9250sensor.cpp (hunk 2 above)
Comment (LETS-BEE) [reply src/sensors/mpu9250sensor.cpp:248]: I didn't code that. that is original code. but I'll fix it.

File: src/sensors/mpu9250sensor.cpp (hunk 3 above)
Comment (LETS-BEE) [reply src/sensors/mpu9250sensor.cpp:170]: oh, I fixed it in my another code. I missed this code. thanks.

**Comment by TheDevMinerTV** [comment]
Please merge this: https://github.com/LETS-BEE/SlimeVR-Tracker-ESP-MPU9250/pull/1

File: src/sensors/mpu9250sensor.cpp (hunk 1 above)
Comment (TheDevMinerTV) [inline src/sensors/mpu9250sensor.cpp:3]: still not fixed

File: src/sensors/mpu9250sensor.cpp [hunk 4]
```diff
@@ -167,15 +166,13 @@ void MPU9250Sensor::getMPUScaled()
 {
     float temp[3];
     int i;
+
+#if defined(_MAHONY_H_) || defined(_MADGWICK_H_)
     int16_t ax,ay,az,gx,gy,gz,mx,my,mz;
```
Comment (TheDevMinerTV) [inline src/sensors/mpu9250sensor.cpp:171]:
```suggestion
    int16_t ax, ay, az, gx, gy, gz, mx, my, mz;
```

File: src/sensors/mpu9250sensor.cpp (hunk 1 above)
Comment (TheDevMinerTV) [inline src/sensors/mpu9250sensor.cpp:3]:
```suggestion
    Copyright (c) 2021 Eiren Rain, S.J. Remington, SlimeVR contributors
```

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

File: src/sensors/mpu9250sensor.cpp (hunk 4 above)
Comment (LETS-BEE) [reply src/sensors/mpu9250sensor.cpp:171]: okay okay. Does the code in the file need to be consistent? I'll fix mpu9250sensors.cpp file

File: src/sensors/mpu9250sensor.cpp (hunk 1 above)
Comment (LETS-BEE) [reply src/sensors/mpu9250sensor.cpp:3]: actually MPU9250 is working from 30/1/2022, #55 . but who didn't care that. so, I think 2022 is right. however I don't know why fix 2021. can you give me more information?

Comment (Eirenliel) [reply src/sensors/mpu9250sensor.cpp:3]: The copyright year is placed either when it's first being made or as a range of when it was made. So it's either 2021 or 2021-2022.

File: src/sensors/mpu9250sensor.cpp (hunk 4 above)
Comment (TheDevMinerTV) [reply src/sensors/mpu9250sensor.cpp:171]: If you copy stuff, you should also fix the style inconsistencies in the code that you are copying.

Comment (LETS-BEE) [reply src/sensors/mpu9250sensor.cpp]: that seems like an implicit rule. I didn't know that. thaks to you.
if i edit or copy the code, i will remember your comment.

**Pull request merged**